File: pset6/readability.py
text = input("Text: ")
l = 0   # Letters
s = 0   # Sentences
w = 0   # Words

for char in text:
    if char.isalnum():
        l += 1
    elif char.isspace():
        w += 1
    elif char == "!" or char == "." or char == "?":
        s += 1

w += 1  # account for extra word at the end

l_w = l / w * 100
s_w = s / w * 100

cl = 0.0588 * l_w - 0.296 * s_w - 15.8

if cl >= 16:
    print("Grade 16+")
elif cl < 1:
    print("Before Grade 1")
else:
    print("Grade ", round(cl))

File: pset2/readability.c
#include <stdio.h>
#include <string.h>
#include <cs50.h>
#include <ctype.h>
#include <math.h>


int main(void)
{
    string text = get_string("Text: ");
    long length = strlen(text);


    long sen = 0;
    long wor = 0;
    long let = 0;
    // loop through and get letters, words, sentences
    for (int c = 0; text[c] != '\0'; c++)
    {
        char ch = tolower(text[c]); // convert to lower case

        if isalnum(ch)
        {
            let += 1;
        }
        else if isspace(ch)
        {
            wor += 1;
        }
        else if (ch == '!' || ch == '?' || ch == '.')
        {
            sen += 1;
            if (c == length - 1)
            {
                wor += 1;
            }
        }
    }
    // printf("Words: %li\n Letters: %li\n Sentences: %li\n", wor,let,sen);

    // convert sen to sen/100 wor, etc.
    double letwor = (double) let / (double) wor * 100;
    double senwor = (double) sen / (double) wor * 100;

    // compute coleman liau
    long cl = roundl(0.0588 * letwor - 0.296 * senwor - 15.8);
    if (cl >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (cl < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %li\n", cl);
    }




}

